cluster_management/src/main/java/com/pinterest/rocksplicator/ConfigGenerator.java

```diff
@@ -237,4 +239,27 @@ private boolean updateDisabledHosts() {
     disabledHosts = latestDisabledInstances;
     return true;
   }
+
+  /**
+   * filter out resources with "Task" state model (ie. workflows and jobs);
+   * only return db resources from ideal states
+   *
+   * db_resource and wf/job have to be unique, bcz wf/job is treated as "resource" in helix.
+   * uniqueness will be enforced when workflow/job/db_resource is added
+   */
+  public List<String> filterOutNoneDbResources(List<String> resources) {
+    HelixAdmin admin = helixManager.getClusterManagmentTool();
+    List<String> db_resources = new ArrayList<>();
+    for (String res : resources) {
+      IdealState ideal = admin.getResourceIdealState(clusterName, res);
+      if (ideal != null) {
+          String stateMode = ideal.getStateModelDefRef();
+          if (stateMode != null && !stateMode.equals("Task")) {
+            db_resources.add(res);
+          }
+      }
+    }
```

Comment:
Can we do it the other way around. i.e., filter out resources that we are sure they are Tasks? This seems to be safer to existing clusters.